lint/doc/Developing.md

```diff
@@ -0,0 +1,28 @@
+# Witnesslint Development Instructions
+
+## Installation for Development
+
+0. You will need a python version >=3.5.
+
+1. Witnesslint uses `poetry` for dependency management and packaging.
+Please refer to https://python-poetry.org/docs/master/#installation for installation instructions.
+
+2. Install project dependencies with `poetry install`.
+
+## Releasing a new version
+
+1. Bump version number in `witnesslint/__init__.py` and `pyproject.toml` to next release version.
+Write an entry for the new version in `doc/Changelog.md`. Commit these changes and nothing else.
+
+2. Build the release archives with `poetry build`. Test the archives and make sure all necessary files are included.
+
+3. Upload the generated archives to PyPI with `poetry publish`. This requires a PyPI account.
+You can use either your username/password combination or an API token for authentication.
+Please refer to `https://python-poetry.org/docs/repositories/#configuring-credentials` for details.
+
+4. Create and push a git tag for the new release.
```

Comment:
maybe it makes sense to do this step (4. pushing the tag) together with (6. bumping to next development version) directly after / when doing 1. . This allows the person performing the release to push all 3 changes (release commit, tag, post-release dev version commit) in one go.
Then 2., 3., and 5. can be done on the release tag without the risk of someone accidentally pushing in the meanwhile (the risk for this is slim, but the proposed change of order also makes it better structured I think)

Reply:
Yes, that seems sensible. It might also make sense to perform step 2 (build and test the relesase archive) first, otherwise we might notice something still missing from the release after pushing the release version/tag.

So the suggestion would be to change the order to `2, 1, 4, 6, 3, 5`.

Reply:
We would only **push**  stuff in 4, 1 I picture as just doing the changes locally

> So the suggestion would be to change the order to 2, 1, 4, 6, 3, 5.

I would bump to the new version (6) before pushing (4), such that we only need to push once, so that would be:  

`2, 1, 6, 4, 3, 5.`

Here we would need to rebuild (2) after doing 1, so as outlined at the beginning of this post I still think it should be:

`1, 2, 6, 4, 3, 5.`

Steps 1,2,6 are local changes and builds, if something goes wrong this can be rewritten. 4 would then push the two commits and the tag to github and make the version official. 3 and 5 is then about publishing the actual release archives. Does that make sense? I think the unclarity stems from the fact that the current description of 1 does not specify whether these changes are done locally or already pushed.

Reply:
Of course we would only push once, in the proposed order `2, 1, 4, 6, 3, 5` step `4` would only be creating the tag, and pushing would be done in step 6. I still think that step `4` should be done before step `6`, as otherwise we have already changed the version number to the next development version when creating the tag.

But I agree that step `2` should be done after `1` after all to avoid having to rebuild for the release. As long as this is done before the push I have no objections. So can we agree on the following: 
```
1, 2, 4, 6, 7, 3, 5
```
where `7` is the push, and `4` is only creating the tag?